features/steps/steps.py

```diff
@@ -134,6 +134,16 @@ def then_i_will_see_on_stdout(context):
     assert_that(context.process.stdout.strip(), equal_to(context.text))
 
 
+@then("if `{value1}` in `{value2}` and stdout matches regexp")
+def then_conditional_stdout_matches_regexp(context, value1, value2):
+    """Only apply regex assertion if value1 in value2."""
+    if value1 in value2.split("||"):
+        assert_that(
```

Comment:
I think we can just call the `then_stdout_matches_regexp`  function here

Reply:
ohh right oops, meant to and then forgot +1